### ISSUE TITLE
ST project import doesn't handle ${workspace_loc}

### ISSUE THREAD

**Issue by Hermann-Core** [microsoft/vscode-embedded-tools#27]
Hi. I'm having theses errors when I import my projrct from STM32CubeIDE. I don't know what is need to be done here
![Capture d’écran 2022-11-05 à 01 06 44](https://user-images.githubusercontent.com/65020150/200093037-25560013-72ba-44cf-b93a-84fbdf399b83.png)

**Comment by benmcmorran** [comment]
Thanks for the report! It looks like this is a bug with our handling of the [`workspace_loc` variable](https://rtist.hcldoc.com/help/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Fconcepts%2Fconcepts-exttools.htm) in STM32CubeIDE projects. I'll work on a fix in the extension, but in the meantime you should be able to work around the issue by editing the paths in the `st-project.cmake` file to replace `${workspace_loc:/${ProjName}/<the rest of the path>}` with `../<the rest of the path>`. For example, the line 22 in your screenshot would become:

```
"$<$<AND:$<CONFIG:Debug>,$<COMPILE_LANGUAGE:ASM>>:${PROJECT_SOURCE_DIR}/build/../Drivers/CMSIS/Core/Include>"
```

**Comment by Hermann-Core** [comment]
> Thanks for the report! It looks like this is a bug with our handling of the [`workspace_loc` variable](https://rtist.hcldoc.com/help/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Fconcepts%2Fconcepts-exttools.htm) in STM32CubeIDE projects. I'll work on a fix in the extension, but in the meantime you should be able to work around the issue by editing the paths in the `st-project.cmake` file to replace `${workspace_loc:/${ProjName}/<the rest of the path>}` with `../<the rest of the path>`. For example, the line 22 in your screenshot would become:
> 
> ```
> "$<$<AND:$<CONFIG:Debug>,$<COMPILE_LANGUAGE:ASM>>:${PROJECT_SOURCE_DIR}/build/../Drivers/CMSIS/Core/Include>"
> ```

Thanks. let me try

**Comment by benmcmorran** [comment]
This issue is fixed in version 0.5.0 of the extension, which is now available in the VS Code Marketplace. The fix should support all paths of the form `${workspace_loc:/${ProjName}/<the rest of the path>}`, and partially support paths of the form `${workspace_loc:/<external project name>/<the rest of the path>}`. Thanks again for your feedback!
